src/noisepy/seis/utils.py

```diff
@@ -0,0 +1,7 @@
+import fsspec
+
+S3PREFIX = "s3"
+
+
+def get_filesystem(path: str) -> fsspec.AbstractFileSystem:
+    return fsspec.filesystem(S3PREFIX, anon=True) if path.startswith(S3PREFIX) else fsspec.filesystem("file")
```

Comment:
This check is not really robust since what if a path is like `s3/mydir/test`? Can't really confirm if it's a url or not. I suggest parsing the path using `urllib.parse` because it will give you an object that will actually parse the url scheme, so you don't need the `S3PREFIX` variable: https://docs.python.org/3/library/urllib.parse.html. Though for windows path, it might not work with `urllib.parse` so you'd need to check for that somehow. 😛

Reply:
Good point, added the check via `urlparse`